cmd/kube-controller-manager/app/options/options.go

```diff
@@ -360,6 +362,13 @@ func (s *KubeControllerManagerOptions) ApplyTo(c *kubecontrollerconfig.Config, u
 	c.Kubeconfig.QPS = s.GenericComponent.KubeAPIQPS
 	c.Kubeconfig.Burst = int(s.GenericComponent.KubeAPIBurst)
 
+	dialer := connrotation.NewDialer((&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext)
```

Comment:
We shouldn't make every client write this block of code, I think--can we do it in the library?

Reply:
@lavalamp The `OnHeartbeatTimeout ` is a handlerFunc when timeout, here I want to close all connections.  

So if we do this in the library, I think we can not support custom timeout handler. And also doing this can influence the client which does not want to use heartbeat.